packages/exporter-trace-otlp-http/src/platform/node/OTLPExporterNodeBase.ts

```diff
@@ -39,7 +39,7 @@ export abstract class OTLPExporterNodeBase<
   DEFAULT_HEADERS: Record<string, string> = {};
   headers: Record<string, string>;
   agent: http.Agent | https.Agent | undefined;
-  compression: CompressionAlgorithm;
+  compression: CompressionAlgorithm | string;
```

Comment:
would be preferable to keep only `CompressionAlgorithm` and correctly parse the env variable to make sure its a valid value

Reply:
This could be accomplished with a typescript type guard like this:

```typescript
function isCompressionAlgorithm(inp?: unknown): inp is CompressionAlgorithm {
  if (typeof inp !== "string") return false;
  if (!algorithms.contains(inp)) return false;
  return true;
}
```

Reply:
> would be preferable to keep only `CompressionAlgorithm` and correctly parse the env variable to make sure its a valid value

@vmarchaud  I added `configureCompression` function to node utils that configures compression (based on config or environment variables) and returns `CompressionAlgorithm`. Also added unit tests for this function.

Reply:
> This could be accomplished with a typescript type guard like this:
> 
> ```ts
> function isCompressionAlgorithm(inp?: unknown): inp is CompressionAlgorithm {
>   if (typeof inp !== "string") return false;
>   if (!algorithms.contains(inp)) return false;
>   return true;
> }
> ```

Just saw this comment @dyladan. Sorry about that. I added a `configureCompression` function that answers @vmarchaud comment. Will that work or should I implement `isCompressionAlgorithm` instead?

Reply:
Looks fine either way